app/handlers/sessions_trusted_launch.rb

```diff
@@ -0,0 +1,32 @@
+class SessionsTrustedLaunch
+
+  lev_handler
+
+  protected
+
+  def authorized?
+    true
+  end
+
+  def handle
+    signup_state = options[:signup_state]
+    uuid_link = UserExternalUuid.find_by_uuid(signup_state.trusted_data['uuid'])
+    if uuid_link.present?
+      outputs[:user] = uuid_link.user
+      signup_state.sign_in!(outputs[:user])
+      return
+    end
+
+    if signup_state.trusted_data['email']
+       user = LookupUsers.by_verified_email(signup_state.trusted_data['email']).first
+       if user
+         UserAlternativeUuid.create(user: user, uuid: trusted_state['uuid'])
+         signup_state.sign_in!(user)
+         outputs[:user] = user
+         return
+       end
+    end
+
```

Comment:
See my other comment on moving code out of `maybe_launch_with_trusted_params` and into this handler...

Here at the bottom, after we have failed to log in an existing user, we could make and store the `SignupState`.  We could also use this spot to encode in the `SignupState` whether we want the user to be asked to choose a password, e.g.

```ruby
outputs.signup_state = 
  if params[:sp][:role] == :instructor
    # Make the SignupState have `require_password` set to true (really `require_authentication` b/c could be social)
  else
    # set `require_password` to false
  end
```
Basically I'm looking for a way to move `after_email_action` out of `SignupState` because that method seems out of the scope of that model (models shouldn't know names of controller actions, but ok in my mind for a model to encode what is logically needed during sign up -- e.g. whether a password is needed or not).

The `handle_with`'s `success` block already takes action based on if `outputs.user` is set.  Similarly, it can use the `outputs.signup_state`'s present or missing email address and `require_authentication` fields to determine where to send the user next (i.e. to collect and verify an email or not, then to collect a password/social auth or not).

With more code in this handler, it will become more of a one-stop shop to see what we do with incoming known users... see if we recognized their external UUID, if so log them in; see if we recognize their external email address, if so log them in; otherwise, send them along the login/signup flow with options set on what we need to collect and what we can skip.